src/main/java/com/xceptance/xlt/report/providers/EventsReportProvider.java

```diff
@@ -68,7 +81,67 @@ public void setConfiguration(ReportProviderConfiguration config)
         super.setConfiguration(config);
 
         // read provider-specific configuration
-        groupEventsByTestCase = ((ReportGeneratorConfiguration) config).getGroupEventsByTestCase();
+        this.groupEventsByTestCase = ((ReportGeneratorConfiguration) config).getGroupEventsByTestCase();
+        this.messageLimit = ((ReportGeneratorConfiguration) config).getEventMessageLimitPerEvent();
+        this.eventLimitPerTestCase = ((ReportGeneratorConfiguration) config).getEventLimitPerTestCase();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void processDataRecord(final Data data)
+    {
+        if (data instanceof EventData)
+        {
+            final EventData eventData = (EventData) data;
+
+            // remember the event
+            eventsPerSecondValueSet.addOrUpdateValue(eventData.getTime(), 1);
+
+            // find the info by test case name
+            Map<String, EventReport> stat = this.testCaseToEventMap.get(eventData.getTestCaseName());
+
+            // unknown?
+            if (stat == null)
+            {
+                // store test case -> event name -> counter and info
+                final EventReport eventReport = new EventReport(eventData.getTestCaseName(), eventData.getName());
+                eventReport.addMessage(eventData.getMessage(), messageLimit);
+
+                stat = new HashMap<>();
+                stat.put(eventData.getName(), eventReport);
+
+                this.testCaseToEventMap.put(eventData.getTestCaseName(), stat);
+            }
+            else
+            {
+                //
+                EventReport eventReport = stat.get(eventData.getName());
+                if (eventReport == null)
+                {
+                    if (stat.size() >= eventLimitPerTestCase)
```

Comment:
Move this check top-level? Currently, if eventLimitPerTestCase is 0, we will get 1 entry nevertheless.